Use enum for vendor ID

davebayer opened the pull request:
This pull request introduces `VkFFTVendor` enum datatype for vendor id instead of using magic values.

davebayer committed:
vendor enum

davebayer committed:
remove vendor enum type base (it is C23 feature)

DTolm commented:
Hello, 

Thank you for the change, it will also make the code more clear. It missed the 0x1027f00 Apple vendor ID definition and I am not sure yet whether the casting of vendor IDs returned by different APIs to the enum will trigger any warnings (in Vulkan I believe it to be an unsigned integer).

Best regards,
Dmitrii

davebayer committed:
fix casts to VkFFTVendor, add VKFFT_VENDOR_APPLE

davebayer commented:
Hi,

I have fixed the casts from different types and added `VKFFT_VENDOR_APPLE` to the enum. I am sorry, I overlooked it completely.

Best regards,
David
